server/routes/api/documents/documents.ts

```diff
@@ -129,7 +130,15 @@ router.post(
       } // otherwise, filter by all collections the user has access to
     } else {
       const collectionIds = await user.collectionIds();
-      where = { ...where, collectionId: collectionIds };
+      where = {
+        ...where,
+        collectionId:
+          template && can(user, "readDocument", user.team)
+            ? {
+                [Op.or]: [{ [Op.in]: collectionIds }, { [Op.is]: null }],
```

Comment:
Lets combine this logic with L103 and guard this else with `!template` to keep the template querying in one place

Reply:
This keeps the `collectionId` logic in one place based on collection-id param from request.
If we move the template condition, we'll have to duplicate the collection auth checks from L117.
I think the current impl provides a single place for `collectionId` related query. Thoughts? 🤔

Reply:
Fair, will reevaluate – I really just dislike the amount of template exceptions that are around now – open to other ideas to make this more long-term maintainable